core/src/main/java/org/elasticsearch/action/admin/indices/delete/TransportDeleteIndexAction.java

```diff
@@ -77,42 +76,23 @@ protected ClusterBlockException checkBlock(DeleteIndexRequest request, ClusterSt
 
     @Override
     protected void masterOperation(final DeleteIndexRequest request, final ClusterState state, final ActionListener<DeleteIndexResponse> listener) {
-        String[] concreteIndices = indexNameExpressionResolver.concreteIndices(state, request);
+        final String[] concreteIndices = indexNameExpressionResolver.concreteIndices(state, request);
         if (concreteIndices.length == 0) {
             listener.onResponse(new DeleteIndexResponse(true));
             return;
         }
         // TODO: this API should be improved, currently, if one delete index failed, we send a failure, we should send a response array that includes all the indices that were deleted
```

Comment:
seems like this TODO can be removed at this point? we have a single cluster state update, so either it fails or it succeeds, no reason to pile up failures for each index anymore?